src/main/java/com/example/school/SchoolApplication.java

```diff
@@ -145,23 +149,85 @@ public static void main(String[] args) {
         group1A.setStudents(students1A);
         group1B.setStudents(students1B);
         group2A.setStudents(students2A);
-        var mark1 = Activity.builder()
+        var activity1 = Activity.builder()
                 .lesson(lesson1)
                 .student(student1)
                 .mark(Optional.of(12))
                 .isPresent(true)
                 .build();
-        var mark2 = Activity.builder()
+        var activity2 = Activity.builder()
                 .lesson(lesson2)
                 .student(student2)
                 .mark(Optional.of(10))
                 .isPresent(true)
                 .build();
-        var mark3 = Activity.builder()
+        var activity3 = Activity.builder()
                 .lesson(lesson3)
                 .student(student7)
                 .mark(Optional.empty())
                 .isPresent(false)
                 .build();
+        var activity4 = Activity.builder()
+                .lesson(lesson1)
+                .student(student2)
+                .mark(Optional.empty())
+                .isPresent(false)
+                .build();
+        var activity5 = Activity.builder()
+                .lesson(lesson3)
+                .student(student2)
+                .mark(Optional.empty())
+                .isPresent(false)
+                .build();
+
+        var activities = List.of(activity1, activity2, activity3, activity4, activity5);
+        System.out.println(getTheWorstSchoolTruant(activities));
+    }
+
+    // method that gets a list of students from the school who skipped the most lessons
+    private static List<Student> getTheWorstSchoolTruant(List<Activity> activities) {
```

Comment:
why do you return list of student, but not one student?

Reply:
I return list of students because it can be several students with the same amount of truants

Reply:
Where should I use the method .collect()? I have already used it in return statement.